### DISCUSSION TITLE
Someone decided to make money on open source

### DISCUSSION THREAD

**Discussion by Kurdakov** [ThioJoe/YT-Spammer-Purge#903] [General]
Found spam in twitter with a link

https://bustinger.com/

obviously someone adjusted code of YTSpammerPurge to run as service and wants to make money.

**Comment by jsdevtom** [comment]
Hi @Kurdakov,

One of the founders here. We've built our own engine, which is faster and more customizable.

We have found this tool and verified it. It had some downsides, bugs, and features missing; therefore we could not use it. It was eating up the quota like it was nothing… impossible to scan in bulk.

**Comment by Kurdakov** [comment]
You are right: there are missing features.

Still quite a strange move: there are no money in this direction.

As for features - It's part of laziness of youtubers (which never came here and offered to finance enhancements - though some start to hire moderators with several thousand dollars pay per month) and users here.
But that can be fixed with relatively small efforts (if only one could face possible block of quota due to use of non API calls): add non API scrapper (for videos and comments as it's done for community posts which was implemented due to my hint ;) ), analyze number of comments under video - if there is increase since previous pass, scan recent comments (not all) (as scammers start to comment recent comments - possibly not most recent, but generally spam posted in 100 recent comments and that is for the reason - scammers afraid that older comments with many likes will be on top and will be marked as spam by regular visitors of video). It will take possibly several days of coding.

If youtube does nothing (but they actually started to make efforts Linus tech tips and JayzTwoCents channels which are part of beta test for new system are free of spam comments), then in near future those features will be added (by me or by some other enterprising individual here)

**Comment by ThioJoe** [comment]
It's unlikely my code could be directly adapted into a web hosted app in any practical way. Like 90% of the code is just "scaffolding" like menus, handling user inputs, configurations, etc. All of which would pretty much be useless and unnecessary if it were hosted in a web interface.

The only thing that might be useful would be the actual filtering logic, but pretty sure you can't copyright ideas anyway, just exact code, and unless they are literally just running it in python on the back end (which wouldn't be very fast on in any scaled operation, like they mentioned) it would have to be re-written anyway.

So basically if someone can decipher my spaghetti and re-write it in a superior package, I guess all the more power to them.

**Comment by ikkysleepy** [comment]
Hey, I want to chime in on this. I created a research tool that has toxicity classifications and has  an UI component and happens to also do some spam classifications. Unfortunately , I found this project after I finished developing my code. I tried to read the spaghetti code but found it to be more trouble then it was worth to try and package it up. Great ideas but this code is not in a modular state where you could make a product without having to heavily refactor the  code.   Great project though.

Here is the link for anyone that is interested in testing out my project:
https://catalina.tinyelectrons.com/

Video Demo:
https://www.youtube.com/watch?v=UJ0z73NazEc

**Reply by ThioJoe** [reply]
> this code is not in a modular state where you could make a product without having to heavily refactor the code

Yea, it's because the origin of this product basically never intended it to become this large. It started out with me literally just wanting a simple script that would delete all comments by a single user on a single video. Had I known I'd be adding so much I would probably have structured and planned it a bit differently. Also, I was effectively learning to code from the very beginning of the project. You should have seen it when it was all one file lol

**Reply by ikkysleepy** [reply]
Yeah, I think that's true for most projects.  As time progresses, you end up cleaning the code / refracting it to be more modular.  There is differently a big difference in what the code is now vs what it could be if knew it would be this big. Anyways, good job on starting this project. Hopefully your coding skills have increased since you started.

**Comment by KendallDoesCoding** [comment]
I'm not sure that's a real product, the website looks too basic tbh. Not a website that would get any customers in my opinion haha
